projects/js-packages/publicize-components/src/components/connection-management/tests/specs/index.test.js

```diff
@@ -0,0 +1,109 @@
+import { screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { getManagementPageObject, setup } from '../factory';
+
+describe( 'ConnectionManagement', () => {
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	describe( 'No connections', () => {
+		test( 'renders the component with no connections', () => {
+			setup( { connections: [] } );
+			const management = getManagementPageObject();
+
+			expect( management.header ).toBeInTheDocument();
+			expect( management.addConnectionButton ).toBeInTheDocument();
+			expect( management.getConnectionByName( 'Facebook' ) ).not.toBeInTheDocument();
+			expect( management.spinners ).toHaveLength( 0 );
+		} );
+	} );
+
+	describe( 'With connections', () => {
+		test( 'renders the spinner without connection name', () => {
+			setup( {
+				connections: [
+					{ service_name: 'twitter', connection_id: '1' },
+					{ service_name: 'facebook', connection_id: '2' },
+				],
+			} );
+			const management = getManagementPageObject();
+
+			expect( management.header ).toBeInTheDocument();
+			expect( management.addConnectionButton ).toBeInTheDocument();
+			expect( management.spinners ).toHaveLength( 2 );
+		} );
+
+		test( 'renders the component with proper connections', () => {
+			setup();
+			const management = getManagementPageObject();
+
+			expect( management.header ).toBeInTheDocument();
+			expect( management.addConnectionButton ).toBeInTheDocument();
+			expect( management.getConnectionByName( 'Twitter' ) ).toBeInTheDocument();
```

Comment:
Technically, it should not render `Twitter` because that should not be returned by the `useSupportedServices` hook.

Reply:
It's mocked here, I can change the names, but I think it shouldn't matter?
https://github.com/Automattic/jetpack/pull/37516/files#diff-4c0ecde683bd576275a34cd0afaa412fd9b74f5cce4693f8e25cd2ab93ccbe07R19

Reply:
Ah OK. That makes sense.